test/rubygems/test_gem_package_tar_writer.rb

```diff
@@ -50,6 +50,17 @@ def test_add_file_source_date_epoch
     end
   end
 
+  def test_add_file_with_mtime
+    Time.stub :now, Time.at(1_458_518_157) do
+      @tar_writer.add_file "x", 0o644, Time.now do |f|
+        f.write "a" * 10
+      end
+
+      assert_headers_equal(tar_file_header("x", "", 0o644, 10, Time.now),
+                         @io.string[0, 512])
```

Comment:
It probably makes sense to store `Time.now` in a variable and use the variable in both cases. It is technically possible for the second to change between the two calls to `Time.now`, and the change is pretty small.

```suggestion
      mtime = Time.now

      @tar_writer.add_file "x", 0o644, mtime do |f|
        f.write "a" * 10
      end

      assert_headers_equal(tar_file_header("x", "", 0o644, 10, mtime),
                         @io.string[0, 512])
```